src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js

```diff
@@ -88,34 +81,38 @@ const convertInclude = async (includeBlock, ...restParams) => {
     return null;
   }
 
-  // This determines if the href has a leading '/'
-  const hrefTypePostion = () => (href.indexOf('/') === 0 ? 1 : 0);
-
-  // This checks if the supportedType is in the correct position of the href
-  const hrefIsSupported = () => supportedType =>
-    href.startsWith(supportedType, hrefTypePostion());
-
-  // This extracts the type from the href
-  const typeExtraction = Object.keys(supportedTypes).find(
-    hrefIsSupported(href),
-  );
+  const { includeType, classification } = includeClassifier({ href, pathname });
 
-  // This determines if the type is supported and returns the include type name
-  const includeType = supportedTypes[typeExtraction];
-  if (!includeType) {
+  if (
+    classification === 'not-supported' ||
+    classification === 'vj-include-not-supporting-amp'
+  ) {
     logger.info(INCLUDE_UNSUPPORTED, {
       type,
+      classification,
       url: href,
     });
     return null;
   }
 
+  let ampSrc;
+  let html;
+  if (classification === 'vj-include-supports-amp') {
+    ampSrc = ampSrcBuilder(href);
+    logger.info(INCLUDE_REQUEST_RECEIVED, {
+      url: ampSrc,
+    });
+  } else {
+    html = await fetchMarkup(buildIncludeUrl(href, includeType, pathname));
+  }
+
```

Comment:
wonder if we could return this implicitly with a tenary
```
  return {
    type,
    model: {
      href,
      type: includeType,
      ...(classification === 'vj-include-supports-amp'
        ? {
            ampSrc: ampSrcBuilder(href),
          }
        : {
            html: await fetchMarkup(
              buildIncludeUrl(href, includeType, pathname),
            ),
          }),
      ...rest,
    },
  };
```

Reply:
this way we won't have `html` or `ampSrc` being undefined in the include block, they won't be there at all

Reply:
Should I move the logging into `ampSrcBuilder`?

Reply:
I think because we `ampSrc` and `html` are `undefined` in the relevant code paths, they are not added to the object, we check for that in our tests:
https://github.com/bbc/simorgh/pull/6873/files#diff-754f58c9944064fefc2f77cd295825f9R325-R336 

I think personally think it is more readable in its current form, but interested to see what others think?

Reply:
> Should I move the logging into ampSrcBuilder? 

I'd suggest leaving logging in the converter as our focused libraries probably shouldn't take on the extra concern of logging, better to keep them [pure](https://medium.com/better-programming/what-is-a-pure-function-3b4af9352f6f) so they don't have any side-effects such as logging.